cloudvolume/datasource/precomputed/mesh/sharded.py

```diff
@@ -0,0 +1,199 @@
+                    mesh.vertices = mesh.vertices.view(dtype=np.int32)
+                    
+                    # Convert from "stored model" space to "model" space
+                    mesh.vertices = manifest.grid_origin + manifest.vertex_offsets[lod] + \
+                                    manifest.chunk_shape * (2 ** lod) * \
+                                    (manifest.fragment_positions[lod][:,frag] + \
+                                    (mesh.vertices / (2.0 ** self.vertex_quantization_bits - 1)))
+
+                    # Scale to native (nm) space
+                    mesh.vertices =  mesh.vertices * (self.transform[0,0], self.transform[1,1], self.transform[2,2])
+                    
+                    lod_meshes.append(mesh)
+
+                # TODO: Fuse meshes before return?
+                meshes.append(lod_meshes)
+
+            results.append(meshes)
+
+        if single_lod_return:
+            results = results[0]
+        if seg_list_return:
+            return results
+        else:
+            return results[0]
+
+class MultiLevelPrecomputedMeshManifest:
+    # Parse the multi-resolution mesh manifest file format:
+    # https://github.com/google/neuroglancer/blob/master/src/neuroglancer/datasource/precomputed/meshes.md
+    # https://github.com/google/neuroglancer/blob/master/src/neuroglancer/mesh/multiscale.ts
+
+    def __init__(self, binary):
+        self._binary = binary
+
+        # num_loads is the 7th word
+        num_lods = int.from_bytes(self._binary[6*4:7*4], byteorder='little', signed=False)
+
+        header_dt = np.dtype([('chunk_shape', np.float32, (3,)),
+                        ('grid_origin', np.float32, (3,)),
+                        ('num_lods', np.uint32),
+                        ('lod_scales', np.float32, (num_lods,)),
+                        ('vertex_offsets', np.float32, (num_lods,3)),
+                        ('num_fragments_per_lod', np.uint32, (num_lods,))
+                        ])
+        self._header = np.frombuffer(self._binary[0:header_dt.itemsize], dtype=header_dt)
+        offset = header_dt.itemsize
+
+        self._fragment_positions = []
+        self._fragment_offsets = []
+        for lod in range(num_lods):
+            # Read fragment positions
+            pos_size =  3 * 4 * self.num_fragments_per_lod[lod]
+            self._fragment_positions.append(
+                np.frombuffer(self._binary[offset:offset + pos_size], dtype=np.uint32).reshape((3,self.num_fragments_per_lod[lod]))
+            )
+            offset += pos_size
+
+            # Read fragment sizes
+            off_size = 4 * self.num_fragments_per_lod[lod]
+            self._fragment_offsets.append(
+                np.frombuffer(self._binary[offset:offset + off_size], dtype=np.uint32)
+            )
+            offset += off_size
+
+        # Make sure we read the entire manifest
+        assert(offset == len(binary))
```

Comment:
I usually prefer not to have asserts in production. Maybe create a new exception or reuse  `cloudvolume.exceptions.MeshDecodeError`?